extensions/blocks/social-previews/upgrade.js

```diff
@@ -0,0 +1,138 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { compact, get } from 'lodash';
+import { Button } from '@wordpress/components';
+import { compose } from '@wordpress/compose';
+import { withDispatch, withSelect } from '@wordpress/data';
+import { addQueryArgs } from '@wordpress/url';
+import { useEffect } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import upgradeImageUrl from './upgrade-illustration.svg';
+import analytics from '../../../_inc/client/lib/analytics';
+import { isSimpleSite } from '../../shared/site-type-utils';
+import getSiteFragment from '../../shared/get-site-fragment';
+
+const SocialPreviewsUpgrade = function SocialPreviewsUpgrade( {
+	autosaveAndRedirect,
+	href,
+	trackViewEvent,
+} ) {
+	// Using the effect here so the tracking is only called once on component mount.
+	useEffect( trackViewEvent, [] );
+
+	return (
+		<div className="jetpack-social-previews__modal-upgrade">
+			<img
+				className="jetpack-social-previews__upgrade-illustration"
+				src={ upgradeImageUrl }
+				width="351"
+				height="264"
+				alt="" // The image is decorative.
+			/>
+			<div className="jetpack-social-previews__upgrade-description">
+				<h2 className="jetpack-social-previews__upgrade-heading">
+					{ __( 'Upgrade to a Business Plan to unlock the power of our SEO tools', 'jetpack' ) }
+				</h2>
+				<ul className="jetpack-social-previews__upgrade-feature-list">
+					<li>
+						{ __(
+							'Preview your site’s content as it will appear on Facebook, Twitter, and the WordPress.com Reader.',
+							'jetpack'
+						) }
+					</li>
+					<li>
+						{ __(
+							'Control how page titles will appear on Google search results and social networks.',
+							'jetpack'
+						) }
+					</li>
+					<li>
+						{ __(
+							'Customize your front page meta data to change how your site appears to search engines.',
+							'jetpack'
+						) }
+					</li>
+				</ul>
+				<Button
+					href={ href } // Only for server-side rendering, since onClick doesn't work there.
+					isPrimary
+					label={ __( 'Purchase a business plan to access social previews', 'jetpack' ) }
+					onClick={ autosaveAndRedirect }
+					target="_top"
+				>
+					{ __( 'Upgrade', 'jetpack' ) }
+				</Button>
+			</div>
+		</div>
+	);
+};
+
+export default compose( [
+	withSelect( select => {
+		const plan = select( 'wordpress-com/plans' ).getPlan( 'business-bundle' );
+		const planPathSlug = get( plan, [ 'path_slug' ] );
+
+		const postId = select( 'core/editor' ).getCurrentPostId();
+		const postType = select( 'core/editor' ).getCurrentPostType();
+
+		// The editor for CPTs has an `edit/` route fragment prefixed.
+		const postTypeEditorRoutePrefix = [ 'page', 'post' ].includes( postType ) ? '' : 'edit';
+
+		// Post-checkout: redirect back here.
+		const redirect_to = isSimpleSite()
```

Comment:
```suggestion
		const redirectTo = isSimpleSite()
```

Minor: also other occurrences of this var need updating.

Reply:
`redirect_to` is the format that WP expects to redirect back to where a user came from after completing the checkout.